multiplatform/greeter/shared/build.gradle.kts

```
@@ -0,0 +1,74 @@
+plugins {
+  alias(libs.plugins.kotlin.multiplatform)
+  alias(libs.plugins.android.library)
+  alias(libs.plugins.ksp)
+}
+
+kotlin {
+  android()
+
+  listOf(
+    iosX64(),
+    iosArm64(),
+    iosSimulatorArm64()
+  ).forEach {
+    it.binaries.framework {
+      baseName = "shared"
+    }
+  }
+
+  sourceSets {
+    val commonMain by getting {
+      dependencies {
+        implementation(libs.kotlinInject.runtime)
+      }
+    }
+    val commonTest by getting {
+      dependencies {
+        implementation(kotlin("test"))
+      }
+    }
+    val androidMain by getting
+    val androidUnitTest by getting
+    val iosX64Main by getting
+    val iosArm64Main by getting
+    val iosSimulatorArm64Main by getting
+    val iosMain by creating {
+      dependsOn(commonMain)
+      iosX64Main.dependsOn(this)
+      iosArm64Main.dependsOn(this)
+      iosSimulatorArm64Main.dependsOn(this)
+    }
+    val iosX64Test by getting
+    val iosArm64Test by getting
+    val iosSimulatorArm64Test by getting
+    val iosTest by creating {
+      dependsOn(commonTest)
+      iosX64Test.dependsOn(this)
+      iosArm64Test.dependsOn(this)
+      iosSimulatorArm64Test.dependsOn(this)
+    }
+  }
+}
+
+android {
+  namespace = "com.fredporciuncula.inject.greeter"
+  compileSdk = 33
+  defaultConfig {
+    minSdk = 24
+    targetSdk = 33
+  }
+  compileOptions {
+    sourceCompatibility = JavaVersion.VERSION_11
+    targetCompatibility = JavaVersion.VERSION_11
+  }
+}
+
+dependencies {
+  // KSP will eventually have better multiplatform support and we'll be able to simply have
+  // `ksp libs.kotlinInject.compiler` in the dependencies block of each source set
+  // https://github.com/google/ksp/pull/1021
+  add("kspIosX64", libs.kotlinInject.compiler)
+  add("kspIosArm64", libs.kotlinInject.compiler)
+  add("kspIosSimulatorArm64", libs.kotlinInject.compiler)
```

And you can loop over iso targets to apply ksp, though idk if there's a better way?
```suggestion
  kotlin.targets.forEach {
    if (it.targetName.contains("ios", ignoreCase = true)) {
      add("ksp${it.targetName.capitalized()}", libs.kotlinInject.compiler)
    }
  }
```

---

Since the future-proof way to go when it comes to iOS targets is to be verbose and explicitly define each iOS target (see [my other comment](https://github.com/evant/kotlin-inject-samples/pull/7#discussion_r1161960130)), I'm wondering if it would make more sense to be verbose here as well. It's definitely annoying to have another place where we have to repeat all iOS targets, though... even though I kinda like how the repetition-prone code is more obvious.

Let me know what you think after going through the other comments, I'm happy to commit your suggestion if you still prefer the loop.

---

Yeah I feel less strongly about it when the targets are explicitly defined. Either way is fine with me honestly.